assembl/fabfile.py

```diff
@@ -2132,6 +2132,56 @@ def set_ssl_certificates():
         print(yellow("Can't set ssl certificates, env.ocsp_path is not set"))
 
 
+@task
+def create_backup_script():
+    """
+    Generates backup script that stores the backup on a local borg repository.
+    Sets a cron job for it.
+    """
+    from jinja2 import Environment, FileSystemLoader
+    print(green("Generating backup script on remote host."))
+    jenv = Environment(loader=FileSystemLoader('doc/'), autoescape=lambda t: False)
+    backup_template = jenv.get_template('backup_template.jinja2')
+    with open('backup_all_assembl.sh', 'w') as f:
+        f.write(backup_template.render(user=env._user, project_path=env._projectpath))
+    put('backup_all_assembl.sh', '/home/assembl_user/backup_all_assembl.sh')
+    run('chmod +x backup_all_assembl.sh')
+    cron_command = "15 3 * * * /home/" + env._user + "/backup_all_assembl.sh"
+    create_add_to_crontab_command(cron_command)
+
+
+@task
+def create_alert_disk_space_script():
+    """Generates the script to alert on disk space limit and sets cron job for it."""
+    from jinja2 import Environment, FileSystemLoader
+    jenv = Environment(loader=FileSystemLoader('doc/'), autoescape=lambda t: False)
+    alert_disk_space_template = jenv.get_template('alert_disk_space_template.jinja2')
+    with open('alert_disk_space.sh', 'w') as f:
+        f.write(alert_disk_space_template.render())
+    put('alert_disk_space.sh', '/home/assembl_user/alert_disk_space.sh')
+    run('chmod +x alert_disk_space.sh')
+    cron_command = "MAILTO=assembl.admin@bluenove.com"
+    run(create_add_to_crontab_command(cron_command))
+    cron_command = "0 5 * * * /home/" + env._user + "/alert_disk_space.sh"
+    run(create_add_to_crontab_command(cron_command))
+
+
+@task
+def install_ncftp_client():
+    """Installs ncftp_client. This client is used to move borg repositories from the
+    local machine to the ftp backup machine or object storage."""
+    print(cyan('Installing ncftp client'))
+    if not env.mac:
```

Comment:
There is a mac version too on brew (`brew install ncftp`), but you don't have to sudo, you can just local it.

Reply:
In any case, I don't know why install this ftp client when each linux/mac comes with `ftp` and `sftp` command. I would assume that you would set up a key with the ftp server created to SSH into it from the database.

Reply:
@ayyazdaniaryan  yeah maybe every mac or linux comes with sftp command. I just followed a tutorial that used this tool so I followed the workflow. Would you like me to keep it, or switch to the default tool ?

Reply:
i was actually inspired by your installation of fail2ban. I figured out that this stuff is only installed on production servers.

Reply:
Yes, let's try to use the tools presented, reducing installation time (and keep docker images small - for docker).